apps/archive/archive.py

```diff
@@ -698,9 +699,30 @@ def get_expired_items(self, expiry_datetime, invalid_only=False):
             query['$and'].append({'expiry_status': {'$ne': 'invalid'}})
 
         req = ParsedRequest()
-        req.max_results = config.MAX_EXPIRY_QUERY_LIMIT
         req.sort = 'expiry,_created'
-        return self.get_from_mongo(req=req, lookup=query)
+        req.where = json.dumps(query)
+        cursor = self.get_from_mongo(req=req, lookup=None)
+        count = cursor.count()
+        no_of_pages = 0
+
+        if count:
+            no_of_pages = len(range(0, count, config.MAX_EXPIRY_QUERY_LIMIT))
+            expiry = cursor[0]['expiry']
+
+        logger.info('Number of items to expire: {}, pages={}'.format(count, no_of_pages))
+
+        for page in range(0, no_of_pages):
```

Comment:
imo it would be better to do `while` loop, it would avoid that initial `count` call that could be expensive, like:
```
while True:
    req = ParsedRequest()
    ...
    items = list(self.get_from_mongo(...))
    if not len(items):
        break
    ...
```
it iterates over all expired items anyhow